### DISCUSSION TITLE
[V3] Collection fail binding if legacy_model_binding is true

### DISCUSSION THREAD

**Discussion by danidoble** [livewire/livewire#6391] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.0-beta.9

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. in config/livewire.php set the `legacy_model_binding` to `true`
2. declare a public collection, after in mount function set some data
```php
public \Illuminate\Database\Eloquent\Collection $users;

public function render()
{
    return view('livewire.users');
}

public function mount(): void
{
    $this->users = \App\Models\User::take(10)->get();
}

public function somethingToTest(): void
{
    if (config('livewire.legacy_model_binding', false)) {
        dd('never comes here');
    } else {
        dd('if legacy_model_binding is false, this will be executed');
    }
}
```

In blade view call function somethingToTest
```blade
<button type="button" wire:click="somethingToTest()">
    Test
</button>
```

And so thrown an exception with `Undefined array key "class"`

In the request i see that class of collection is not sent
```json
{"data":{"users":[[],{"s":"elcl"}]},"memo":{"id":"yg9qJao9x9XTIyp7THiZ","name":"users","path":"\/","method":"GET","children":[],"lazyLoaded":true,"errors":[],"locale":"en"},"checksum":"248c7e11f3153a6ffabe778c4336c9fb879a3256814fb0119ab88e2dedac2a17"}
```

## Edit

In `EloquentCollectionSynth.php` in method `dehydrate` if comment 
```php 
if (empty($rules)) return [[], []];
```
works, but i don't know for what is there that condition

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by luanfreitasdev** [comment, marked as answer]
Added this Pull Request 
https://github.com/livewire/livewire/pull/6441
